app/scripts/modules/core/src/state/recoilAtoms.ts

```diff
@@ -0,0 +1,7 @@
+import { atom } from 'recoil';
+import { CollapsibleSectionStateCache } from '../cache/collapsibleSectionStateCache';
+
+export const verticalNavExpandedAtom = atom({
```

Comment:
I think we should put atom definitions closer to where they will be used.  A global `state` package doesn't smell right to me.   @vigneshm @caseyhebebrand @erikmunson thoughts?

Reply:
I am personally a fan of a centralized state, especially considering that the point of atoms is to be used in multiple places. 

 This atom will be read from the navigation, but written to in the header component.  So I could put this atom with the `verticalNav` since the key references that.

Reply:
Open to the majority opinion. When working with redux, it was nice to have all the selectors, reducers, etc in one folder since many times they were not tied to one particular scope of the app or data.

Reply:
> I am personally a fan of a centralized state

I am _very strongly_ opposed to a redux-like global centralized state in spinnaker.  The reason recoiljs appeals to me is that we can share state between arbitrary component trees without introducing a centralized global state.

Reply:
I spent sometime working on a react state management framework and I _think_ (although not sure) `recoiljs` may have been influenced by some of the design decisions.

The framework's primary focus was to let domain owners define their own state and let it be shared across react component trees. Usually the modules related to a domain such as react components, utilities, pages etc are co-located and this framework let state management related modules to be co-located as well. This makes it possible to automatically route bugs/runtime errors by looking at the stack trace and finding the owner of the module (via some metadata associated with each module).

Another advantage of splitting state management is that it allows lazy loading of modules. If there's some page (or domain) of an application that is not frequently visited, then you can load the related modules on-demand when the page is requested. With centralized state, all the state initializers and mutators/reducers would be loaded during application bootstrap even though many may not be needed or you would need to explicitly manage the lazy loading of these modules.  

I generally prefer splitting state management for these reasons.